src/controller/cna.controller/cna.controller.js

```diff
@@ -35,231 +35,249 @@ async function getCna (req, res) {
-    }
-
-    if (req.body.name.surname !== undefined) {
-      if (typeof req.body.name.surname !== 'string') {
-        return res.status(400).json({ message: 'Wrong value type' })
+      if (!result) {
+        logger.info('The user could not be created because ' + cnaShortName + ' CNA does not exist.')
+        return res.status(400).json({ message: 'The user could not be created because ' + cnaShortName + ' CNA does not exist.' })
       }
 
-      user.name.surname = req.body.name.surname
-    }
+      const key = uuidAPIKey.create()
+      user.secret = key.uuid
+      const newUser = new User(user)
+
+      // Find user in MongoDB
+      User.findOne()
+        .byUserNameAndCnaShortName(newUser.username, cnaShortName)
+        .exec((err, result) => {
+          if (err) {
+            logger.warn('An error ocurred: ' + newUser.username + ' was not found in the database.')
+            logger.error(err.stack)
+            return res.status(500).send('Internal Server Error')
+          }
+
+          // Create user in MongoDB if it doesn't exist
+          if (!result) {
+            User.findOneAndUpdate()
+              .byUserNameAndCnaShortName(newUser.username, cnaShortName)
+              .updateOne(newUser)
+              .setOptions({ upsert: true })
+              .exec((err) => {
+                if (err) {
+                  logger.error(err.stack)
+                  return res.status(500).send('Internal Server Error')
+                }
+
+                newUser.secret = key.apiKey
+                const responseMessage = {
+                  message: 'The user was successfully created.',
+                  created: newUser
+                }
+
+                logger.info(JSON.stringify(responseMessage))
+                return res.status(200).json(responseMessage)
+              })
+          } else {
+            logger.info(newUser.username + ' was not created because it already exists.')
+            return res.status(400).json({ message: newUser.username + ' already exists.' })
+          }
+        })
+    })
+}
 
-    if (req.body.name.suffix !== undefined) {
-      if (typeof req.body.name.suffix !== 'string') {
-        return res.status(400).json({ message: 'Wrong value type' })
-      }
+// TODO: Request body sanitation and parameter validation
+// Updates a user only if the user exist for the specified username. If no user exists, it does not create the user.
+async function updateUser (req, res) {
+  const username = req.params.username
+  const cnaShortName = req.params.shortname
+  const user = req.body
 
-      user.name.suffix = req.body.name.suffix
-    }
+  if (req.body.username !== undefined) {
+    user.username = req.body.username
+  } else {
+    user.username = username
   }
 
-  const key = uuidAPIKey.create()
-  user.secret = key.uuid
+  if (req.body.cna_short_name !== undefined) {
+    user.cna_short_name = req.body.cna_short_name
+  } else {
+    user.cna_short_name = cnaShortName
+  }
```

Comment:
Change to:

if (user.username === undefined) {
    user.username = username
  }

  if (user.cna_short_name === undefined) {
    user.cna_short_name = cnaShortName
  }